### PR TITLE
Fix Task component tests

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Some tests were failing silently due to bad structure, e.g. tests
not waiting for a promise to be resolved or not making any assertions.

Split the tests for the 'select default step' behaviour to assert
that the `onSelect` callback is called with the expected values for
task id and step id, and separately test that the `selectedStepId`
prop is processed correctly. This approach avoids having to use additional
test-specific wrapper components to handle the callback and provide
the updated prop value and keeps the purpose of the tests more clear.

Thanks to @LyndseyBu for catching this.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: packages/components/src/components/Task/Task.test.js [hunk 1]
```diff
@@ -129,31 +145,40 @@ describe('Task', () => {
...
+    const stepName = 'a step';
+    const steps = [{ name: stepName }];
+    const { getByText } = render(
+      <Task {...props} expanded selectedStepId={stepName} steps={steps} />
     );
+
+    expect(
+      getByText(
+        (content, element) =>
+          content === 'a step' &&
```
Comment (briangleeson) [inline packages/components/src/components/Task/Task.test.js:157]: Can use `content === stepName &&` here

**Review by briangleeson — COMMENTED** [review]
/lgtm

One nit but looks good

File: packages/components/src/components/Task/Task.test.js (hunk 1 above)
Comment (AlanGreene) [reply packages/components/src/components/Task/Task.test.js]: good catch, updated

**Pull request merged**